### PR TITLE
changes required for NFV adoption

### PR DESCRIPTION
**Pull request by eshulman2**
- Re-factored the way the openstack command is foremed for dev env
- Moved static code to vars
- Changed hosts for minimal job to use localhost instead of local which will allow avoiding using an inventory with just localhost in it

### CODE REVIEW

File: tests/vars.sample.yaml
Comment (jistr) [inline tests/vars.sample.yaml:4]: Should we have these two src_env_bastion variables provided as defaults, or is it ok leaving them out from here? Maybe going forward we should prefer overriding the `openstack_ssh_command` (or `source_openstack_command`) as a whole, without going through the intermediate variables... It kinda evolved organically and we need them to not break existing jobs, but i don't know if it's the best pattern to keep expanding on...

File: tests/vars.sample.yaml
Comment (frenzyfriday) [inline tests/vars.sample.yaml:7]: I think this should be "openstack_ssh_command"

File: tests/roles/development_environment/tasks/main.yaml
Comment (frenzyfriday) [inline tests/roles/development_environment/tasks/main.yaml:7]: I think this should be "openstack_ssh_command"

File: tests/roles/development_environment/tasks/main.yaml
Comment (frenzyfriday) [inline tests/roles/development_environment/tasks/main.yaml:16]: I think this should be "openstack_ssh_command"

File: tests/roles/development_environment/tasks/main.yaml
Comment (frenzyfriday) [inline tests/roles/development_environment/tasks/main.yaml:23]: also here

**Comment by eshulman2** [comment]
recheck

**Comment by eshulman2** [comment]
recheck

**Comment by jistr** [comment]
Looks like an issue coming from the refactoring rather than CI instability:

```
"+ secret store --name testSecret --payload TestPayload", "/bin/bash: line 4: secret: command not found"
```

File: tests/roles/backend_services/tasks/main.yaml
Comment (marios) [inline tests/roles/backend_services/tasks/main.yaml:100]: had to stare at this for a bit. removing the kustomization tls_overrides.j2 and merging it into the openstack_control_plane. 
i just question if it is really necessary to do it in this patch where you have lots of other nfv specific/required things going on.
It is the kind of change that might benefit of its own review.

for example, does it even make sense to keep a config/base directory any more then? this file is now no longer used https://github.com/marios/data-plane-adoption/blob/main/tests/config/base/openstack_control_plane.yaml lets remove that one too (i.e. you're now always generating it from the j2)

**Review by marios — COMMENTED** [review]
fyi the last run on the jobs here is hitting a known issue https://issues.redhat.com/browse/OSPCIX-504 

(eg see https://logserver.rdoproject.org/92/592/20e171faa4189ed80e4ee9bc44e080eaeba567e4/github-check/adoption-standalone-to-crc-ceph/51ac3f1/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackdataplanedeployments.dataplane.openstack.org/openstack.yaml)
```
    - lastTransitionTime: "2024-09-25T20:41:52Z"
      message: Deployment error occurred in ovn service error ConfigMap "ovncontroller-config"
        not found
```

**Comment by jistr** [comment]
recheck

**Comment by eshulman2** [comment]
recheck

**Comment by jistr** [comment]
I think this failed on https://issues.redhat.com/browse/OSPCIX-503 which should now be worked around, will try another recheck.

**Comment by jistr** [comment]
recheck

**Review by jistr — APPROVED** [review]
LGTM i'm just wondering if we should have a testproject for one of the multinode jobs before we merge.

File: tests/roles/backend_services/tasks/main.yaml
Comment (jistr) [reply tests/roles/backend_services/tasks/main.yaml:100]: FWIW my instinct is the same re making this a separate PR, but i'm also thinking about timeframes and letting other patches rebase after these changes are made, so if this ends up working ok in CI or just needing minor adjustments then perhaps our best path could be to merge the "big things" fast and iterate on any leftovers / polishing?

Comment (cescgina) [reply tests/roles/backend_services/tasks/main.yaml:100]: I agree, ideally it would be best to have this change into 2 or 3, but realistically, we should merge this when we get green runs for multinode jobs. I have a small PR that will need to be rebased on top of this to solve conflicts, I can include the cleanup of these unused files there

**Review by cescgina — APPROVED** [review]

**Pull request merged**
